chain/client/src/sync.rs

```diff
@@ -486,9 +482,7 @@ impl BlockSync {
         if header_head.height <= head.height {
             return Ok(vec![]);
         }
-        let close_to_header_head =
-            last_header.height() > header_head.height.saturating_sub(self.block_fetch_horizon);
-        if self.cache.len() < BLOCK_SYNC_CACHE_LOWER_LIMIT || close_to_header_head {
+        if (self.cache.len() as u64) < self.block_fetch_horizon {
```

Comment:
@SkidanovAlex I think this will make the node more susceptible to eclipse attacks. To address this, we can attempt to check whether the last header is on the same chain as header head, but I don't think it's worth spending the time to polish it, given that we will likely change this code completely.